src/main/java/net/dries007/tfc/network/PacketSimpleMessage.java

```diff
@@ -0,0 +1,134 @@
+package net.dries007.tfc.network;
+
+import java.nio.charset.Charset;
+
+import io.netty.buffer.ByteBuf;
+import net.dries007.tfc.ConfigTFC;
+import net.dries007.tfc.TerraFirmaCraft;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.Style;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler;
+import net.minecraftforge.fml.common.network.simpleimpl.MessageContext;
+import stanhebben.zenscript.util.ArrayUtil;
+
+public class PacketSimpleMessage implements IMessage
```

Comment:
Rather than this rather convoluted and difficult to maintain serialization, it would be much better to use vanilla's built-in handling for this. `ITextComponent` has a serialize to JSON, which we can read + write to a string, which can be written to a packet. Much better than individually trying to serialize every text component field.